securesystemslib/signer.py

```diff
@@ -137,8 +150,264 @@ def to_dict(self) -> Dict:
+    """Public  GPG key.
+
+    Provides a verify method to verify a cryptographic signature with a
+    gpg-style rsa, dsa or ecdsa public key on the instance.
+
+    Note that this implementation is not TUF or in-toto spec compliant
+
+    Attributes:
+        type: Key type, e.g. "rsa", "dsa" or "ecdsa".
+        method: GPG Key Scheme
+        hashes: list of GPG Hash Algorithms, e.g. "pgp+SHA2".
+        keyval: Opaque key content.
+        keyid: Key identifier
+        creation_time: Unix timestamp when GPG key was created.
+        validity_period: Validity of the GPG Keys in days.
+        subkeys: A dictionary containing keyid and GPG subkey.
+    """
+
+    type: str
+    method: str
+    hashes: List[str]
+    keyval: Dict[str, str]
+    keyid: str
+    creation_time: Optional[int] = None
+    validity_period: Optional[int] = None
+    subkeys: Optional[Dict[str, "GPGKey"]] = None
+
+    @classmethod
+    def from_dict(cls, keyid: str, key_dict: Dict[str, Any]) -> "GPGKey":
+        """Creates ``GPGKey`` object from its json/dict representation.
+        Raises:
+            KeyError, TypeError: Invalid arguments.
+        """
+        subkeys_dict = key_dict.get("subkeys")
+
+        gpg_subkeys = None
+        if subkeys_dict:
+            gpg_subkeys = {
+                subkeyid: GPGKey.from_dict(subkeyid, subkey_dict)
+                for (subkeyid, subkey_dict) in subkeys_dict.items()
+            }
+
+        return cls(
+            key_dict["type"],
+            key_dict["method"],
+            key_dict["hashes"],
+            key_dict["keyval"],
+            key_dict["keyid"],
+            key_dict.get("creation_time"),
+            key_dict.get("validity_period"),
+            gpg_subkeys,
+        )
+
+    def to_dict(self):
+        """Returns the dictionary representation of self."""
+
+        key_dict = {
+            "method": self.method,
+            "type": self.type,
+            "hashes": self.hashes,
+            "keyid": self.keyid,
+            "keyval": self.keyval,
+        }
+
+        if self.creation_time:
+            key_dict["creation_time"] = self.creation_time
+        if self.validity_period:
+            key_dict["validity_period"] = self.validity_period
+        if self.subkeys:
+            subkeys_dict = {
+                keyid: subkey.to_dict()
+                for (keyid, subkey) in self.subkeys.items()
+            }
+            key_dict["subkeys"] = subkeys_dict
+
+        return key_dict
+
+    def verify_signature(self, signature: Signature, data: bytes) -> None:
+        if not gpg.verify_signature(signature.to_dict(), self.to_dict(), data):
+            raise exceptions.UnverifiedSignatureError(
+                f"Failed to verify signature by {self.keyid}"
+            )
+
+
+# SecretsHandler is a function the calling code can provide to Signer:
+# If Signer needs secrets from user, the function will be called
+SecretsHandler = Callable[[str], str]
+
+
 class Signer:
```

Comment:
Signer class definition should probably include public_key as an attribute now: 
* the rest of the API (like from_priv_key_uri()) kind of assumes Signer always has a Key
* implementing e.g. Signer creation/import becomes more straight forward if the key is somehow available Example:
    ```
    # Load an existing KMS key, import the public key from it
    signer = GCPSigner.import(gcp_key_id)
    pubkey = signer.public_key
    ```

Reply:
I am not exposing Signer.public_key or making it a required construction arg for SSllibSigner in this PR. 

I think both probably make sense (for e.g. the key import use case) but the PR is large already and does not itself require those changes: will file further issues.